sensors/simsensor_common/simsensor_reader.c

```diff
@@ -0,0 +1,273 @@
+{
+	size_t lineLen = getline(&rd->lineBuf, &rd->bufLen, rd->scenarioFile);
+	if (lineLen < 1) {
+		return -1;
+	}
+
+	/* Checking if current line is header */
+	if (rd->lineBuf[0] == 's' || rd->lineBuf[0] == 'S') {
+		return 0;
+	}
+
+	if (fseek(rd->scenarioFile, -lineLen, SEEK_CUR) != 0) {
+		return -1;
+	}
+
+	clearerr(rd->scenarioFile);
+
+	return 0;
+}
+
+
+int reader_open(simsens_reader_t *rd, const char *path, int sensorTypes, time_t timeHorizon)
+{
+	if (rd == NULL || path == NULL) {
+		return -1;
+	}
+
+	rd->scenarioFile = fopen(path, "r");
+	if (rd->scenarioFile == NULL) {
+		return -1;
+	}
+
+	rd->lineBuf = NULL;
+
+	rd->bufLen = 0;
+	rd->sensorTypes = sensorTypes;
+	rd->timeHorizon = timeHorizon;
+
+	rd->timeLast = READER_TIMESTAMP_NOT_SET;
+	rd->offset = 0;
+
+	if (reader_skipHeader(rd) != 0) {
+		fclose(rd->scenarioFile);
+		free(rd->lineBuf);
+		return -1;
+	}
+
+	return 0;
+}
+
+
+void reader_close(simsens_reader_t *reader)
+{
+	if (reader == NULL) {
+		return;
+	}
+
+	if (reader->scenarioFile != NULL) {
+		fclose(reader->scenarioFile);
+	}
+
+	free(reader->lineBuf);
+}
+
+
+static int reader_getFieldLLong(const char *actField, char **nextField, long long *res)
+{
+	char *endptr;
+	long long value = strtoll(actField, &endptr, 10);
+
+	if (!(*endptr == READER_DATA_SEPARATOR || *endptr == '\0' || *endptr == '\n')) {
+		return -1;
+	}
+
+	if (*endptr == READER_DATA_SEPARATOR) {
+		*nextField = endptr + 1;
+	}
+
+	*res = value;
+
+	return 0;
+}
+
+
+static int reader_accelDataParse(char *startOfDataSection, time_t timestamp, sensor_event_t *result)
+{
+	char *actField, *nextField;
+	long long tmp;
+
+	actField = startOfDataSection;
+
+	result->type = SENSOR_TYPE_ACCEL;
+	result->timestamp = timestamp;
+
+	if (reader_getFieldLLong(actField, &nextField, &tmp) < 0) {
+		return -1;
+	}
+	result->accels.accelX = tmp;
+	actField = nextField;
+
+	if (reader_getFieldLLong(actField, &nextField, &tmp) < 0) {
+		return -1;
+	}
+	result->accels.accelY = tmp;
+	actField = nextField;
+
+	if (reader_getFieldLLong(actField, &nextField, &tmp) < 0) {
+		return -1;
+	}
+	result->accels.accelZ = tmp;
+	actField = nextField;
+
+	return 0;
+}
+
+
+static int reader_baroDataParse(char *startOfDataSection, time_t timestamp, sensor_event_t *result)
+{
+	char *actField, *nextField;
+	long long tmp;
+
+	actField = startOfDataSection;
+
+	result->type = SENSOR_TYPE_BARO;
+	result->timestamp = timestamp;
+
+	if (reader_getFieldLLong(actField, &nextField, &tmp) < 0) {
+		return -1;
+	}
+	result->baro.pressure = tmp;
+	actField = nextField;
+
+	if (reader_getFieldLLong(actField, &nextField, &tmp) < 0) {
+		return -1;
+	}
+	result->baro.temp = tmp;
+	actField = nextField;
+
+	return 0;
+}
+
+
+int reader_read(simsens_reader_t *rd, event_queue_t *queue)
+{
+	char *actField, *nextField;
+	long long tmp;
+	int sensorID, err = 0, remainEmptyInter = 100;
+	size_t curLineLen;
+	time_t timestamp, first_timestamp = READER_TIMESTAMP_NOT_SET;
+	sensor_event_t parsedResult;
```

Comment:
```suggestion
	int sensorID, err = 0, emptyIter = 100;
	size_t lineLen;
	time_t timestamp, timeStart = READER_TIMESTAMP_NOT_SET;
	sensor_event_t parsed;
```
 - `first_timestamp` does not follow coding convention
 - `curLineLen`, `remainEmptyInter` and `parsedResult` are a little too long. I gave propositions on shortening them without loosing the meaning. What do you think about these?